### DISCUSSION TITLE
Hello There 👋

### DISCUSSION THREAD

**Discussion by ahmedsaed** [ahmedsaed/Ahmedsaed#3] [General]
```
  ___      _     ___  ___                  
 / _ \    | |    |  \/  |                  
/ /_\ \___| | __ | .  . | ___              
|  _  / __| |/ / | |\/| |/ _ \             
| | | \__ \   <  | |  | |  __/             
\_| |_/___/_|\_\ \_|  |_/\___|             
                                           
                                           
  ___              _   _     _             
 / _ \            | | | |   (_)            
/ /_\ \_ __  _   _| |_| |__  _ _ __   __ _ 
|  _  | '_ \| | | | __| '_ \| | '_ \ / _` |
| | | | | | | |_| | |_| | | | | | | | (_| |
\_| |_/_| |_|\__, |\__|_| |_|_|_| |_|\__, |
              __/ |                   __/ |
             |___/                   |___/ 
```

Feel free to ask your doubts in this discussion.

**Comment by yann013** [comment]
Hello Ahmed I have a question actually

**Reply by yann013** [reply]
if tou have a discord please allow me to add you

**Reply by ahmedsaed** [reply]
Sure, here is my username `ahmedsaed`

**Reply by yann013** [reply]
Thanks you

**Reply by yann013** [reply]
If ever it's faster here, so here I saw your script for an Ai car with NavMesh, I tried to do it with That and waypoints which gave me the following script:
```cs
using System.Collections;
using System.Collections.Generic;
using UnityEngine;
using UnityEngine.AI;

public class AiVoit : MonoBehaviour
{
    [SerializeField] private Chemin chemin;
    [SerializeField] private WheelData[] wheels;
    public float vitAcc = 50.0f; //vitesse accel
    public float freinAcc = 90.0f;
    public float virage = 1.05f;
    public float capRotVolant = 35f;
    public Light[] lumiereFrein;

    private NavMeshAgent agent;
    private Transform posChemin;

    public float distRay = 10.0f; 
    public Transform[] RayPoint;


    [System.Serializable]
    public class WheelData
    {
        public WheelCollider collider;
        public MeshRenderer mesh;
    }

    void Start()
    {
        agent = GetComponent<NavMeshAgent>();
        posChemin = chemin.prochLigne(null);
        agent.SetDestination(posChemin.position);
    }

    void Update()
    {
        if (agent.remainingDistance < 0.1f)
        {
            posChemin = chemin.prochLigne(posChemin);
            agent.SetDestination(posChemin.position);
            transform.Rotate(0f, 180f, 0f);
        }

        UpdateRouePos();
        volant();
        frein();

        // Check for obstacles using raycasts
        bool obstacleDetected = false;
        foreach (Transform point in RayPoint)
        {
            RaycastHit hit;
            if (Physics.Raycast(point.position, transform.forward, out hit, distRay))
            {
                obstacleDetected = true;
                Debug.DrawRay(point.position, transform.forward * hit.distance, Color.red, 0.1f); // Visualize raycast
                break; // Stop checking if an obstacle is found
            }
        }

        // Modify car behavior based on obstacle detection
        if (obstacleDetected)
        {
            // Slow down, change direction, etc.
            agent.speed = agent.speed * 0.5f; // Example: Reduce speed
        }
    }

    void volant()
    {
        Vector3 direction = agent.steeringTarget - transform.position;
        float nvVolant = Mathf.Clamp(Vector3.SignedAngle(transform.forward, direction, Vector3.up), -capRotVolant, capRotVolant);

        foreach (var wheel in wheels)
        {
            if (wheel.collider.transform.localPosition.z < 0) //permet de faire tourner les roue du devant 
            {
                var rotVolant = nvVolant * virage;

                wheel.collider.steerAngle = rotVolant;
            }
        }
    }

    void frein()
    {
        if (agent.velocity.magnitude < 0.1f)
        {
            foreach (var wheel in wheels)
            {
                wheel.collider.brakeTorque = 500 * freinAcc * Time.deltaTime;
            }

            foreach (Light brakeLight in lumiereFrein)
            {
                brakeLight.intensity = 1f;
            }
        }
        else
        {
            foreach (var wheel in wheels)
            {
                wheel.collider.brakeTorque = 0;
            }
            foreach (Light brakeLight in lumiereFrein)
            {
                brakeLight.intensity = 0f;
            }
        }
    }

    private void UpdateRouePos()
    {
        float bougeDetails = -Mathf.Clamp(agent.velocity.magnitude, -1, 1);
        foreach (WheelData wheel in wheels)
        {
            wheel.collider.motorTorque = bougeDetails * 600 * vitAcc * Time.deltaTime;

            Quaternion quat;
            Vector3 position;
            wheel.collider.GetWorldPose(out position, out quat);
            wheel.mesh.transform.position = position;
            wheel.mesh.transform.rotation = quat;
        }
    }
}
```

It works except that he doesn't really respect it and does a bit of anything to date, the script that really works is him

```cs
using System.Collections;
using System.Collections.Generic;
using UnityEngine;
using UnityEngine.AI;

public class AiVoit : MonoBehaviour
{

    [SerializeField] private Chemin chemin;
    [SerializeField] private float vitDepl = 5f;
    [SerializeField] private float distancePause = 0.5f;

    private Transform posChemin;
    // Start is called before the first frame update
    void Start()
    {

        //Va envoyer la voiture au premier point
        posChemin = chemin.prochLigne(posChemin);
        transform.position = posChemin.position;

        //proch point
        posChemin = chemin.prochLigne(posChemin);
        transform.LookAt(posChemin);
    }

    // Update is called once per frame
    void Update()
    {

        transform.position = Vector3.MoveTowards(transform.position, posChemin.position, vitDepl * Time.deltaTime);
        if (Vector3.Distance(transform.position, posChemin.position) < distancePause)
        {
            posChemin = chemin.prochLigne(posChemin);
            transform.LookAt(posChemin);
        }
    }
}

```

**Reply by ahmedsaed** [reply]
It appears you're attempting to utilize the carAI script. While I'd genuinely like to assist you, I must clarify that I discontinued this project two years ago, and actually moved on from game development entirely. Therefore, I can't provide additional support beyond what's detailed in the project documentation.

Although, If you have any specific questions, feel free to ask me and I will try answer them

**Reply by yann013** [reply]
Oh, okay, no problem. Well, thank you for your answer, but I want to know something: when you wrote that code, did you record it?

**Reply by ahmedsaed** [reply]
I didn't capture a code walkthrough, but I did create a tutorial demonstrating how to utilize the package.

https://youtu.be/_8_g2xWi7XU

**Reply by yann013** [reply]
hey bro do you understand django-python?

**Reply by ahmedsaed** [reply]
No, I haven't worked with that framework before.

**Reply by yann013** [reply]
oh alright no problem

**Comment by wzzim** [comment]
Hey Ahmed, would like to know why you prefer Linux over Windows.  And how did you get through the learning curve? Any vids or inspiration for me to have a better understanding of Linux? Cuz im genuinely interested. Thanks!

**Reply by ahmedsaed** [reply]
Well, it's mostly a matter of preference but once I started using Linux as a daily driver, I began appreciating the freedom and the control you get. I am also a big fan of Gnome. it's workflow is wonderful and I like it's minimalism and stability. Also as a developer, Linux is far superior as a development environment than windows unless your are developing windows applications or games. 

In my opinion, the best way to learn Linux is just to install it and use it as your main OS. Force yourself into adopting it's workflow and methodology. You will face lots of challenges which will teach you a ton about Linux and operating systems in general. It's more of a learning journey than just an OS. You also have to be prepared to spend a lot of time tinkering and fixing stuff at the beginning.

One of the cool things about Linux is it's community. If you have any questions, feel free to ask me or on any linux community and don't get discouraged by the ignorant people.

**Reply by wzzim** [reply]
Ok well, I just did my research. I heard Arch is very tough but I'm on summer break right now and it's worth the challenge + it has the most customization so that's pretty neat.
Here are my specs:
Device name	Wassim (Hp envy 13 - ad1xx)
Processor	Intel(R) Core(TM) i7-8550U CPU @ 1.80GHz   1.99 GHz
GPU                 Intel (R) UHD Graphics 620
Installed RAM	8.00 GB (7.85 GB usable)
System type	64-bit operating system, x64-based processor
Storage            300/500gb free space

I would like to know if it's worth it to Dual boot and get to choose between Windows & Linux whenever I want. And if coding and other multi-tasking tasks will be way easier on linux and load faster

**Reply by ahmedsaed** [reply]
I wouldn't recommend arch if it's your first time unless you really want to explore how operating systems are structured under the hood. If you are planning to use the linux system for other things, I would recommend a more stable distro. Something that's not a rolling release like arch. I personally love fedora but you can also use ubuntu, mint and others.

Linux is lightweight. You don't have to worry about your specs. 

Regarding the dual boot thing. I am against it unless you really rely on software that is not supported on linux and doesn't have alternatives. Having windows as a fall back will not force you into learning the differences in linux. You will learn a ton by exploring and debugging the issues you encounter.
